test/parallel/test-http2-compat-serverrequest.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+// Http2ServerRequest should expose convenience properties
+
+const server = h2.createServer();
+server.listen(0, common.mustCall(function() {
+  server.once('request', common.mustCall(function(request, response) {
+    const expected = {
+      statusCode: null,
+      version: '2.0',
+      httpVersionMajor: 2,
+      httpVersionMinor: 0
+    };
+
+    assert.strictEqual(request.statusCode, expected.statusCode);
+    assert.strictEqual(request.httpVersion, expected.version);
+    assert.strictEqual(request.httpVersionMajor, expected.httpVersionMajor);
+    assert.strictEqual(request.httpVersionMinor, expected.httpVersionMinor);
+
+    response.stream.on('finish', common.mustCall(function() {
+      server.close();
+    }));
+    response.end(' ');
+  }));
+
+  const url = `http://localhost:${server.address().port}`;
+  const client = h2.connect(url, common.mustCall(function() {
+    const headers = {
+      ':path': '/foobar',
+      ':method': 'GET',
+      ':scheme': 'http',
+      ':authority': 'localhost'
+    };
+    const request = client.request(headers);
+    request.on('end', common.mustCall(function() {
+      client.socket.destroy();
```

Comment:
Note: Once https://github.com/nodejs/http2/pull/64 lands, this can be simplified to just `client.destroy()`